widgets/nuxeo-uploader-behavior.html

```diff
@@ -102,9 +109,163 @@
         }
         return operation.input(this.uploader)
           .params(params)
-          .execute(options)
+          .execute(options);
+      }.bind(this));
+    };
+
+    /**
+     * Returns whether the provider accepts a given file for upload or not.
+     *
+     * @param {Object} file the file to be checked
+     * @return {Boolean} true if the current file is accepted by the current provider, false otherwise
+     **/
+    DefaultUploadProvider.prototype.accepts = function(file) {
+      var mimeType = ((file.type !== '') ? file.type.match(/^[^\/]*\//)[0] : null);
+      var fileType = file.name.match(/\.[^\.]*$/)[0];
+
+      return !(this.accept && !(this.accept.indexOf(mimeType) > -1 || this.accept.indexOf(fileType) > -1));
+    };
+
+    /**
+     * Returns whether or not the provider reports file upload progress.
+     *
+     * @return {Boolean} true if the provider reports progress, false otherwise
+     **/
+    DefaultUploadProvider.prototype.hasProgress = function() {
+      return false;
+    };
+
+    var _registry = { 'default': DefaultUploadProvider };
+    var _defaultProvider = 'default';
+
+    /**
+     * @polymerBehavior Nuxeo.UploaderBehavior
+     */
+    Nuxeo.UploaderBehavior = {
+
+      properties: {
+        /**
+         * Accepted file extensions or mime types (comma separated values).
+         */
+        accept: String,
+
+        /**
+         * This flag determines whether the file should be immediately uploaded or not.
+         */
+        immediate: {
+          type: Boolean,
+          value: true
+        },
+
+        /**
+         * Current batch id.
+         */
+        batchId: String,
+
+        /**
+         * List of files in the current batch.
+         */
+        files: {
+          type: Array,
+          value: []
+        },
+
+        /**
+         * Flag that indicates if an upload is in progress.
+         */
+        uploading: {
+          type: Boolean,
+          value: false
+        },
+
+        /**
+         * Allow multiple files to be added to the same batch.
+         */
+        batchAppend: {
+          type: Boolean,
+          value: false
+        },
+
+        provider: {
+          type: String,
+          value: function() {
+            return _defaultProvider;
+          }
+        }
+      },
+
+      observers: [ '_initProvider(provider, connection, accept, batchAppend)'],
+
+      _initProvider: function() {
+        if (this.provider) {
+          this._providerInstance = new _registry[this.provider](this.connection, this.accept, this.batchAppend);
+        }
+      },
+
+      _reloadDefaultProvider: function() {
```

Comment:
is it worth to add this method? couldn't you just do this in the method below? ⬇️